File: actin-common/src/main/java/com/hartwig/actin/clinical/curation/config/PrimaryTumorConfig.java
package com.hartwig.actin.clinical.curation.config;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PrimaryTumorConfig implements CurationConfig {

    @NotNull
    @Override
    public abstract String input();

    @NotNull
    public abstract String primaryTumorLocation();

    @NotNull
    public abstract String primaryTumorSubLocation();

    @NotNull
    public abstract String primaryTumorType();

    @NotNull
    public abstract String primaryTumorSubType();

    @NotNull
    public abstract String primaryTumorExtraDetails();

    @NotNull
    public abstract Set<String> doids();
}

File: clinical/src/main/java/com/hartwig/actin/clinical/curation/config/PrimaryTumorConfig.kt
package com.hartwig.actin.clinical.curation.config

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class PrimaryTumorConfig : CurationConfig {
    abstract override fun input(): String
    override fun ignore(): Boolean {
        return false
    }

    abstract fun primaryTumorLocation(): String
    abstract fun primaryTumorSubLocation(): String
    abstract fun primaryTumorType(): String
    abstract fun primaryTumorSubType(): String
    abstract fun primaryTumorExtraDetails(): String
    abstract fun doids(): Set<String?>
}